clash-lib/src/Clash/Core/Evaluator/Models.hs

```diff
@@ -232,27 +213,26 @@ updateGlobal i x =
     Nothing ->
       pure ()
 
--- | Remove a global term binding from the environment, then evaluate the given
--- action. The removed binding is restored after evaluating the action.
---
-withoutGlobal :: Id -> Eval a -> Eval a
-withoutGlobal i x = RWS.modify' ignoreBinding >> x
- where
-  ignoreBinding env =
-    let is = genvIgnore env
-     in env { genvIgnore = Set.insert i is }
+getRecInfo :: Eval RecInfo
+getRecInfo = RWS.gets genvRecInfo
+
+getFuel :: Eval Word
+getFuel = RWS.gets genvFuel
+
+putFuel :: Word -> Eval ()
+putFuel x = RWS.modify' (\env -> env { genvFuel = x })
 
 -- | Run the given action, preserving the state of the global ignore list
 -- before the action was run. This means any global identifiers which are
 -- ignored can be seen again when continuing evaluation.
 --
-preserveIgnored :: Eval a -> Eval a
-preserveIgnored x = do
-  is  <- RWS.gets genvIgnore
-  res <- x
-  env <- RWS.get
+preserveState :: Eval a -> Eval a
+preserveState x = do
+  fuel <- RWS.gets genvFuel
+  res  <- x
+  env  <- RWS.get
 
-  RWS.put (env { genvIgnore = is })
+  RWS.put (env { genvFuel = fuel })
```

Comment:
Is it a good idea to call this `preserveState` when it only preservers fuel?

Reply:
I changed it to `preserveState` when it had both fuel and the ignore list, I guess `preserveFuel` is a better name for the end result though. I don't think anything else would need to be saved / restored later...